components/layout_2020/dom_traversal.rs

```diff
@@ -299,7 +298,7 @@ impl Drop for BoxSlot<'_> {
 pub(crate) trait NodeExt<'dom>: 'dom + Copy + LayoutNode + Send + Sync {
     fn is_element(self) -> bool;
     fn as_text(self) -> Option<String>;
-    fn as_image(self) -> Option<(Option<Arc<NetImage>>, Vec2<Length>)>;
+    fn as_image(self) -> Option<(Option<Arc<NetImage>>, Vec2<f64>)>;
```

Comment:
Why the type change, and why not `CSSFloat`?

Reply:
It’s not `Length` because it is measured in image pixels rather than CSS `px`.

I wanted to make it `u32` but it’s already divided by `script::dom::htmlimageelement::ImageRequest::current_pixel_density`. I’m not sure what that density means exactly or if diving there is correct.

It could be `CSSFloat`. It probably wouldn’t affect much since it’s not kept in memory for a long time.